doc/secure_boot/hld_secure_boot.md

```diff
@@ -0,0 +1,332 @@
+# HLD Secure Boot #
+
+##  1. <a name='TableofContent'></a>Table of Content
+<!-- vscode-markdown-toc -->
+* 1. [Table of Content](#TableofContent)
+	* 1.1. [Revision](#Revision)
+	* 1.2. [Scope](#Scope)
+	* 1.3. [Definitions/Abbreviations](#DefinitionsAbbreviations)
+	* 1.4. [Overview](#Overview)
+	* 1.5. [Requirements](#Requirements)
+	* 1.6. [Architecture Design](#ArchitectureDesign)
+	* 1.7. [High-Level Design](#High-LevelDesign)
+		* 1.7.1. [Flow diagram](#Flowdiagram)
+		* 1.7.2. [Module Elements Breakdown](#ModuleElementsBreakdown)
+		* 1.7.3. [Sign Flow diagram description:](#SignFlowdiagramdescription:)
+		* 1.7.4. [Runtime SB Verification FLOW:](#RuntimeSBVerificationFLOW:)
+	* 1.8. [SAI API](#SAIAPI)
+	* 1.9. [Configuration and management](#Configurationandmanagement)
+		* 1.9.1. [CLI/YANG model Enhancements](#CLIYANGmodelEnhancements)
+		* 1.9.2. [Config DB Enhancements](#ConfigDBEnhancements)
+	* 1.10. [Warmboot and Fastboot Design Impact](#WarmbootandFastbootDesignImpact)
+	* 1.11. [Restrictions/Limitations](#RestrictionsLimitations)
+	* 1.12. [Testing Requirements/Design](#TestingRequirementsDesign)
+		* 1.12.1. [Unit Test cases](#UnitTestcases)
+		* 1.12.2. [System Test cases](#SystemTestcases)
+	* 1.13. [Open/Action items - if any](#OpenActionitems-ifany)
+
+<!-- vscode-markdown-toc-config
+	numbering=true
+	autoSave=true
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+###  1.1. <a name='Revision'></a>Revision
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 06/2022 |   David Pilnik   | Phase 1 Design     |
+###  1.2. <a name='Scope'></a>Scope
+
+This secure boot HLD doc described the requirements, architecture, and configuration details of the secure boot feature in switches Sonic OS based.
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations
+	SB - Secure Boot
+	KO - Kernel Objects
+
+
+###  1.4. <a name='Overview'></a>Overview
+
+Secure Boot is a feature found in the startup software for your computer that's designed to ensure your computer starts safely and securely by preventing unauthorized software like malware from taking control of your Device at boot-up.
+
+The entire point is the "chain of trust" it creates. If I have a piece of software that I need to make sure that nothing malicious can intercept what I am doing and put their own code in I need to trust the program launching my program.
+
+What Secure boot provides is an anchor for that first "trusted program". It allows the hardware on the computer to assert "No one has modified this bootloader and it will behave exactly like the original programmers programed". The boot-loader can then check "No one has modified this OS and it will behave exactly like the original programmers programmed".(Chain of trust described below)
```

Comment:
since we talked about this chain of trust, i do not understand why this HLD leave the discussion of root of trust out of the scope. then, how would this feature be leveraged by the community?

Reply:
In this reviewed HLD, the root of of trust is the BIOS(UEFI).

(can be seen in diagram 1.7.1 - chapter name: Run-time verification flow diagram.)
So vendors can store their public keys in the UEFI DB, by doing that, they enforce to boot images that were only signed with a private key that matches the vendor pub key stored.

Reply:
Does this HLD address TPM (Trusted Platform Modules) for SONiC?

Reply:
> since we talked about this chain of trust, i do not understand why this HLD leave the discussion of root of trust out of the scope. then, how would this feature be leveraged by the community?

@lguohan  - If the question is about HW RoT it is platform-dependent. On Intel Platform it can be the Intel BootGuard which is the Field-Upgradable Fuse (FPF) in PCH with the hash of the OEM Public key. For ARM - it is about some secure registers/fuses that also keep the hash of the public key and the BL1 of ARM Trusted Firmware (ATF). These HW RoT validate the first components of the UEFI framework starting the Chain-of-Trust. Since these solutions are very much platform-dependent they are left out of scope of the current HLD

Reply:
> Does this HLD address TPM (Trusted Platform Modules) for SONiC?

Nope